src/shims/windows/foreign_items.rs

```diff
@@ -838,64 +1170,99 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 // It's fine to not use the Handle type here because its a stub
                 this.write_int(1, dest)?;
             }
             "GetModuleHandleA" if this.frame_in_std() => {
-                #[allow(non_snake_case)]
-                let [_lpModuleName] =
-                    this.check_shim_sig_lenient(abi, sys_conv, link_name, args)?;
+                let [module_name] = this.check_shim_sig(
+                    shim_sig!(extern "system" fn(*const _) -> winapi::HMODULE),
+                    link_name,
+                    abi,
+                    args,
+                )?;
                 // We need to return something non-null here to make `compat_fn!` work.
                 this.write_int(1, dest)?;
             }
             "SetConsoleTextAttribute" if this.frame_in_std() => {
-                #[allow(non_snake_case)]
-                let [_hConsoleOutput, _wAttribute] =
-                    this.check_shim_sig_lenient(abi, sys_conv, link_name, args)?;
+                let [console_output, attribute] = this.check_shim_sig(
+                    shim_sig!(extern "system" fn(winapi::HANDLE, u16) -> winapi::BOOL),
+                    link_name,
+                    abi,
+                    args,
+                )?;
                 // Pretend these does not exist / nothing happened, by returning zero.
                 this.write_null(dest)?;
             }
             "GetConsoleMode" if this.frame_in_std() => {
-                let [console, mode] =
-                    this.check_shim_sig_lenient(abi, sys_conv, link_name, args)?;
+                let [console, mode] = this.check_shim_sig(
+                    shim_sig!(extern "system" fn(winapi::HANDLE, *mut _) -> winapi::BOOL),
+                    link_name,
+                    abi,
+                    args,
+                )?;
                 this.read_target_isize(console)?;
                 this.deref_pointer_as(mode, this.machine.layouts.u32)?;
                 // Indicate an error.
                 this.write_null(dest)?;
             }
             "GetFileType" if this.frame_in_std() => {
-                #[allow(non_snake_case)]
-                let [_hFile] = this.check_shim_sig_lenient(abi, sys_conv, link_name, args)?;
+                let [file] = this.check_shim_sig(
+                    shim_sig!(extern "system" fn(winapi::HANDLE) -> u32),
+                    link_name,
+                    abi,
+                    args,
+                )?;
                 // Return unknown file type.
                 this.write_null(dest)?;
             }
             "AddVectoredExceptionHandler" if this.frame_in_std() => {
-                #[allow(non_snake_case)]
-                let [_First, _Handler] =
-                    this.check_shim_sig_lenient(abi, sys_conv, link_name, args)?;
+                let [first, handler] = this.check_shim_sig(
+                    shim_sig!(extern "system" fn(u32, *mut _) -> *mut _),
```

Comment:
I standardized all the names. I can undo this is desired, it just seemed like a good drive-by since most stuff uses Rust-style names, not the Windows API names.